client/task-list/tasks/payments/RemotePayments/index.js

```diff
@@ -73,6 +67,13 @@ export const RemotePayments = ( { query } ) => {
 		} );
 	};
 
+	const [ enabledMethods, setEnabledMethods ] = useState(
+		methods.reduce( ( acc, method ) => {
+			acc[ method.key ] = method.isEnabled;
```

Comment:
I think this may need to be updated whenever `methods` changes since `methods` won't be preloaded and immediately available on page load.

Reply:
You're right, I've made the change in ffd49ed to reactively change `enabledMethods`. After reading the rest of the code, it seems like there are more refactors to be made, however, I feel is a bit out of scope from this PR.